src/ExtensionDownloader.ts

```diff
@@ -103,3 +120,16 @@ export class ExtensionDownloader
         return success;
     }
 }
+
+function readParserVersionFromFile(logger: Logger) {
+    const basePath = util.getExtensionPath();
+    const absolutePath = path.resolve(basePath, 'parser-version.txt');
+
+    try {
+        const data = fs.readFileSync(absolutePath, { encoding: 'utf8' });
+        return data;
+    } catch (error) {
+        logger.appendLine('No parser-version.txt file found.')
+        return '';
+    }
+}
```

Comment:
Note: https://stackoverflow.com/questions/44704968/visual-studio-code-insert-newline-at-the-end-of-files will add a trailing newline as a common practice and prevent this little symbol in GitHub